diff -ru a/tests/test-different-number-streams.c b/tests/test-different-number-streams.c
--- a/tests/test-different-number-streams.c	2021-04-14 01:32:15.803282771 +0000
+++ b/tests/test-different-number-streams.c	2021-04-14 01:31:56.906289893 +0000
@@ -73,6 +73,7 @@
   guint ls_id, rs_id_1, rs_id_2;
   gchar *lufrag = NULL, *lpassword = NULL;
   gchar *rufrag1 = NULL, *rpassword1 = NULL, *rufrag2 = NULL, *rpassword2 = NULL;
+  NiceAddress *addr;
 
 #ifdef G_OS_WIN32
   WSADATA w;
@@ -80,12 +81,18 @@
   WSAStartup(0x0202, &w);
 #endif
 
+  /* Initialize nice agents */
+  addr = nice_address_new ();
+  nice_address_set_from_string (addr, "127.0.0.1");
+
   global_mainloop = g_main_loop_new (NULL, FALSE);
 
   /* step: create the agents L and R */
   lagent = nice_agent_new (g_main_loop_get_context (global_mainloop),
       NICE_COMPATIBILITY_GOOGLE);
   g_debug ("lagent: %p", lagent);
+
+  nice_agent_add_local_address (lagent, addr);
   nice_agent_set_software (lagent, "test-different-number-streams, Left Agent");
   g_object_set (G_OBJECT (lagent), "ice-tcp", FALSE,  NULL);
   g_object_set (G_OBJECT (lagent), "controlling-mode", TRUE, NULL);
@@ -98,6 +105,8 @@
   ragent = nice_agent_new (g_main_loop_get_context (global_mainloop),
       NICE_COMPATIBILITY_GOOGLE);
   g_debug ("ragent: %p", ragent);
+
+  nice_agent_add_local_address (ragent, addr);
   nice_agent_set_software (ragent, "test-different-number-streams, Right Agent");
   g_object_set (G_OBJECT (ragent), "ice-tcp", FALSE,  NULL);
   g_object_set (G_OBJECT (ragent), "controlling-mode", FALSE, NULL);
